api/v2/views/base.py

```diff
@@ -0,0 +1,153 @@
+        instance = self.get_object()
+        if (hasattr(instance, "is_closed")
+            and instance.is_closed()):
+            message = (
+                "Method '%s' not allowed: "
+                "the request has already been resolved."
+                % self.request.method
+            )
+            raise exceptions.MethodNotAllowed(self.request.method,
+                                              detail=message)
+        else:
+            return fn(self, request, *args, **kwargs)
+    return wrapper
+
+
+class BaseRequestViewSet(viewsets.ModelViewSet):
+    """
+    Base class ViewSet to handle requests
+    """
+    admin_serializer_class = None
+
+    model = None
+
+    def get_queryset(self):
+        """
+        Return users requests or all the requests if the user is an admin.
+        """
+        assert self.model is not None, (
+            "%s should include a `model` attribute."
+            % self.__class__.__name__
+        )
+        if self.request.user.is_staff:
+            return self.model.objects.all()
+        return self.model.objects.filter(created_by=self.request.user)
+
+    def get_serializer_class(self):
+        """
+        Return the `serializer_class` or `admin_serializer_class`
+        given the users privileges.
+        """
+        assert self.admin_serializer_class is not None, (
+            "%s should include an `admin_serializer_class` attribute."
+            % self.__class__.__name__
+        )
+        if self.request.user.is_staff:
+            return self.admin_serializer_class
+        return self.serializer_class
+
+    def perform_create(self, serializer):
+        # NOTE: An identity could possible have multiple memberships
+        # It may be better to directly take membership rather than an identity
+        identity_id = serializer.initial_data("identity")
+        status, _ = StatusType.objects.get_or_create(name="pending")
+        try:
+            membership = IdentityMembership.objects.get(identity=identity_id)
+            instance  = serializer.save(
+                membership=membership,
+                status=status,
+                created_by=self.request.user
+            )
+            self.submit_action(instance)
+        except core_exceptions.ProviderLimitExceeded:
+            message = "Only one active request is allowed per provider."
+            raise exceptions.MethodNotAllowed('create', detail=message)
+        except core_exceptions.InvalidMembership:
+            message = (
+                "The user '%s' is not a valid member."
+                % self.request.user.username
+            )
+            raise exceptions.ParseError(detail=message)
+        except IdentityMembership.DoesNotExist:
+            message = (
+                "The identity '%s' does not have a membership"
+                % identity_id
+            )
+            raise exceptions.ParseError(detail=message)
+
+    @unresolved_requests_only
```

Comment:
Yep. Use functools.wraps() to ensure mutliple decorators work. They are evaluated top down. But if you don't use classes and wraps, it may not compose properly.